src/ImageSharp/Formats/Png/PngScanlineProcessor.cs

```diff
@@ -128,32 +120,24 @@ internal static class PngScanlineProcessor
 
             if (header.BitDepth == 16)
             {
-                Rgba64 rgba64 = default;
                 for (int x = pixelOffset, o = 0; x < header.Width; x += increment, o += 2)
                 {
                     ushort luminance = BinaryPrimitives.ReadUInt16BigEndian(scanlineSpan.Slice(o, 2));
-                    rgba64.R = luminance;
-                    rgba64.G = luminance;
-                    rgba64.B = luminance;
-                    rgba64.A = luminance.Equals(luminance16Trans.PackedValue) ? ushort.MinValue : ushort.MaxValue;
+                    ushort alpha = luminance.Equals(luminance16Trans.PackedValue) ? ushort.MinValue : ushort.MaxValue;
```

Comment:
`alpha` is known outside the loop, doing the comparison each time is redundant.

Reply:
It's not known unfortunately, we're comparing the current luminance to the trans value.